### PR TITLE
Bugfix/lattice 897 es pt table write creation synch

### PR DESCRIPTION
**Pull request by you-leee**

### CODE REVIEW

File: src/main/java/com/openlattice/datastore/services/EdmService.java [hunk 1]
```diff
@@ -469,18 +474,20 @@ public void createEntitySet( Principal principal, EntitySet entitySet, Set<UUID>
...
                         authorizations.setSecurableObjectType( aclKey,
                                 SecurableObjectType.PropertyTypeInEntitySet );
-                    } ).forEach( aclKey -> authorizations.addPermission(
-                    aclKey,
-                    principal,
-                    EnumSet.allOf( Permission.class ) ) );
+                    } )
+                    .forEach( aclKey -> authorizations.addPermission( aclKey, principal, EnumSet.allOf( Permission.class ) ) );
+
+            List<PropertyType> ownablePropertyTypes = Lists.newArrayList( propertyTypes.getAll( ownablePropertyTypeIDs ).values() );
```
Comment (geekbeast) [inline src/main/java/com/openlattice/datastore/services/EdmService.java]: Is there a reason you need to copy from Collection into List?

File: src/main/java/com/openlattice/datastore/services/EdmService.java [hunk 2]
```diff
@@ -926,11 +933,16 @@ public void updatePropertyTypeMetadata( UUID propertyTypeId, MetadataUpdate upda
...
                             .forEach( es -> eventBus
                                     .post( new PropertyTypesInEntitySetUpdatedEvent( es.getId(), properties ) ) );
                 } );
-        eventBus.post( new PropertyTypeCreatedEvent( getPropertyType( propertyTypeId ) ) );
+
+        PropertyType propertyType = getPropertyType( propertyTypeId );
+        edmManager.createPropertyTypeIfNotExist( propertyType );
+        esEdmService.createPropertyType( propertyType );
+
+        eventBus.post( new PropertyTypeCreatedEvent( propertyType ) );
```
Comment (geekbeast) [inline src/main/java/com/openlattice/datastore/services/EdmService.java]: Their should be a PropertyTypeUpdateEvent, but outside of the scope of this PR.

File: src/main/java/com/openlattice/datastore/services/EdmService.java [hunk 3]
```diff
@@ -926,11 +933,16 @@ public void updatePropertyTypeMetadata( UUID propertyTypeId, MetadataUpdate upda
...
                             .newArrayList( propertyTypes.getAll( et.getProperties() ).values() );
-                    entitySetManager.getAllEntitySetsForType( et.getId() )
+                    edmManager.getAllEntitySetsForType( et.getId() )
                             .forEach( es -> eventBus
                                     .post( new PropertyTypesInEntitySetUpdatedEvent( es.getId(), properties ) ) );
                 } );
-        eventBus.post( new PropertyTypeCreatedEvent( getPropertyType( propertyTypeId ) ) );
+
+        PropertyType propertyType = getPropertyType( propertyTypeId );
+        edmManager.createPropertyTypeIfNotExist( propertyType );
```
Comment (geekbeast) [inline src/main/java/com/openlattice/datastore/services/EdmService.java]: I'm not sure if its checked earlier, but this won't have any effects as the tables will already exists in Postgres. You need to see if it is one the allowed changes and then issue the appropriate ALTER TABLE statements.

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java [hunk 4]
```diff
@@ -143,68 +161,102 @@ public void revoke(
...
     }
 
-    private String grantOnTable( String table, String principalId, String permission ) {
-        return String.format( "GRANT %s ON TABLE %s TO %s", permission, table, principalId );
-    }
-
-    private String renameColumn( String table, String current, String update ) {
-        return String.format( "ALTER TABLE %s RENAME COLUMN %s TO %s", table, current, update );
-    }
+    public Iterable<EntitySet> getAllEntitySets() {
```
Comment (geekbeast) [inline src/main/java/com/openlattice/edm/PostgresEdmManager.java]: This should be switched to using PostgresIterable, but is outside the scope of this PR.

File: src/main/java/com/openlattice/edm/PostgresEdmManager.java [hunk 5]
```diff
@@ -143,68 +161,102 @@ public void revoke(
...
-     * all include an if not exists. If the behavior of that class changes this function should be updated
-     * appropriately.
-     */
-    @VisibleForTesting
-    public void createPropertyTypeTableIfNotExist( EntitySet entitySet, PropertyType propertyType )
-            throws SQLException {
-        PostgresTableDefinition ptd = DataTables.buildPropertyTableDefinition( entitySet, propertyType );
-        ptm.registerTables( ptd );
-    }
+    public Iterable<EntitySet> getAllEntitySetsForType( UUID entityTypeId ) {
```
Comment (geekbeast) [inline src/main/java/com/openlattice/edm/PostgresEdmManager.java]: PostgresIterable.

File: src/main/java/com/openlattice/datastore/services/EdmService.java (hunk 1 above)
Comment (you-leee) [reply src/main/java/com/openlattice/datastore/services/EdmService.java]: In the ElasticSearchApi the saveEntitySetToElasticsearch is declared with a List as input, which is called from EsEdmService

**Pull request merged**